src/main.jsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable react/prop-types */
+
+import React from "react"
+import ReactDOM from "react-dom"
+import { AutoplayVideo } from "./lib"
+
+function App() {
+  const [videoRef, setVideoRef] = React.useState()
+
+  return (
+    <>
+      <AutoplayVideo
+        ref={setVideoRef}
+        src="/example-assets/example-video.mp4"
+        posterImg="/example-assets/example-fallback.jpg"
+        description="This is a description of the video."
+        width={16}
+        height={9}
+        prefersReducedMotion={false}
+        renderReducedMotionFallback={() => (
+          <img
+            src="/example-assets/example-fallback.jpg"
+            alt="Description of the fallback image."
+          />
+        )}
+      />
+
+      <DebugStats videoElement={videoRef} />
+    </>
+  )
+}
+
+/**
+ * The purpose of this debug snipe is to provide a tool to ensure that the
+ * IntersectionObserver aspect of the video component is working. The video's
+ * currentTime property will be displayed, and should only be incrementing
+ * when the video is considered to be "in view" (per the ovbserver options).
+ */
+function DebugStats({ videoElement }) {
+  const rafId = React.useRef()
+  const timeRef = React.useRef()
+
+  const updateStats = React.useCallback(() => {
+    if (!videoElement && rafId.current !== undefined) {
+      cancelAnimationFrame(rafId.current)
+      return
+    }
+
+    timeRef.current.innerHTML = videoElement.currentTime
+    requestAnimationFrame(updateStats)
+  }, [videoElement])
+
+  React.useEffect(() => {
+    rafId.current = requestAnimationFrame(updateStats)
```

Comment:
probs not necessary given its just for debugging but asking out of curiosity - because the dependency is `updateStats` which checks for the existence of the `videoElement` and conditionally calls `cancelAnimationFrame` is that why we don't need a cleanup `return` here?

Reply:
I think that was my thinking @rvno, but it's probably cleaner to run it in a cleanup instead. Good catch

Reply:
changed that to what you suggested — fixes the render issue with the demo too 😎  thanks